### PR TITLE
checkstyle: ensure curly brackets have whitespace around them #181

### LINKED ISSUE #181: checkstyle: ensure curly brackets have whitespace around them
There are many instances of code in the code base where the curly brackets (`{}`) are not separated by whitespace:

        protected void raise(BaseEvent event){
        -------------------------------------^
                            Space required before '{'

However, checkstyle does not catch it.

### PR DESCRIPTION
**Pull request by pyokagan**
Fixes #181 

The Java coding standard[1] requires that curly brackets ({}) be
surrounded by whitespace.

However, there are several violations of this requirement in the code
base. e.g.

    protected void raise(BaseEvent event){
    -------------------------------------^
                        Space required before '{'

Fix all of these code style violations.

Our checkstyle configuration failed to catch these errors because we did
not tell checkstyle to check for the LCURLY, RCURLY and SLIST (statement
list) tokens. Fix this in the checkstyle config file so that such code
style violations will not happen again.

[1] https://oss-generic.github.io/process/codingstandards/coding-standards-java.html#methods

### CODE REVIEW

**Comment by damithc** [comment]
@m133225 reminder to review

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -215,11 +215,21 @@
...
         LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF, LITERAL_RETURN,
         LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE, LOR, LT, MINUS,
         MINUS_ASSIGN, MOD, MOD_ASSIGN, NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION,
-        SL, SL_ASSIGN, SR_ASSIGN, STAR, STAR_ASSIGN"/>
+        RCURLY, SL, SLIST, SL_ASSIGN, SR_ASSIGN, STAR, STAR_ASSIGN"/>
+      <!-- Allow empty constructors e.g. MyClass() {} -->
+      <property name="allowEmptyConstructors" value="true" />
+      <!-- Allow empty methods e.g. void func() {} -->
+      <property name="allowEmptyMethods" value="true" />
+      <!-- Allow empty types e.g. class Foo{}, enum Foo {} -->
```
Comment (m133225) [inline config/checkstyle/checkstyle.xml]: Should have a space before `{` in the class example. 😛

**Comment by m133225** [comment]
Looks good aside from the minor issue commented on!

**Comment by pyokagan** [comment]
@m133225 v2 ready for review

**Comment by m133225** [comment]
Looks good.

**Pull request merged**